File: Daily Test/C++/Minimum Moves To Open Lock 11.4.2025.cpp
#include <iostream>
#include <string>
#include <cmath>
using namespace std;

int main() {
    string s1, s2;
    cin >> s1 >> s2;
    int total_moves = 0;

    for (int i = 0; i < s1.length(); i++) {
        int d1 = s1[i] - '0';
        int d2 = s2[i] - '0';
        int clockwise = abs(d1 - d2);
        int anticlockwise = 10 - clockwise;
        total_moves += min(clockwise, anticlockwise);
    }

    cout << total_moves << endl;
    return 0;
}

File: Daily Test/Java/Minimum Moves To Open Lock 11.4.2025.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s1 = sc.nextLine().trim();
        String s2 = sc.nextLine().trim();
        int total_moves = 0;

        for (int i = 0; i < s1.length(); i++) {
            int d1 = s1.charAt(i) - '0';
            int d2 = s2.charAt(i) - '0';
            int clockwise = Math.abs(d1 - d2);
            int anticlockwise = 10 - clockwise;
            total_moves += Math.min(clockwise, anticlockwise);
        }

        System.out.println(total_moves);
    }
}